packages/records/lib/models/records.ts

```diff
@@ -703,39 +705,39 @@ export async function update({
  * @param environmentId - The id of the environment.
  * @param connectionId - The id of the connection.
  * @param model - The model name.
+ * @param mode - The deletion mode: 'hard' (permanent deletion), 'soft' (empty payload and set deleted_at) or 'prune' (empty payload only).
  * @param limit - The maximum number of records to delete.
  * @param toCursorIncluded - The cursor up to which records should be deleted (inclusive. ordered by updated_at, id).
  * @param batchSize - The number of records to delete in each batch (default is 1000).
- * @param mode - The deletion mode: 'hard' (permanent deletion) or 'soft' (empty payload and set deleted_at). Default is 'hard'.
  */
 export async function deleteRecords({
     connectionId,
     environmentId,
     model,
+    mode,
     limit,
     toCursorIncluded,
-    batchSize = 1000,
-    mode = 'hard'
+    batchSize = 1000
 }: {
     connectionId: number;
     environmentId: number;
     model: string;
+    mode: 'hard' | 'soft' | 'prune';
```

Comment:
For now soft delete, hard delete, prune are disjoint options of the deleteRecords function but could make it a bitwise parameter (or similar) if we need to combine them in the future.